in-progress/7520-testnet-overview.md

```diff
@@ -0,0 +1,391 @@
+   - Storage requirements typically measured in GBs (< 1TB)
+   - Synchronization time is typically measured in hours/days.
+3. **Archive Node**:
+   - Receive and validate blocks (header and body) from the p2p network
+   - Stores the full state history of the chain
+   - Stores all blocks since genesis
+   - Can respond to queries of state at any point in time
+   - Storage requirements typically measured in TBs
+   - Synchronization time is typically measured in hours/days.
+
+### Open Questions
+
+- What kind of "watcher" role can full nodes play?
+
+## Prover Nodes
+
+Prover nodes will receive information from proposers and will be responsible for creating proofs, and posting them to L1.
+
+### Open Questions
+
+- What is the interface that proposers will use to communicate with prover nodes?
+  - Do they need one other than L1?
+
+## Proposer/Validator Selection
+
+As noted above, the initial Rollup contract will allow holders of TST to stake a set amount of their tokens to become validators.
+
+Much of Ethereum will be mimicked in that one user can have multiple validators.
+
+A randao will be used to select a committee from the validator set for each epoch.
+
+Each slot in an epoch will be randomly assigned to a validator in the committee.
+
+### Open Questions
+
+- How resistant is this to censorship?
+- How expensive is it to run?
+- How can reward be distributed?
+- What are the slashing conditions?
+- Probability/severity/cost of different attacks based on the power of attacker (1%, 5%, 10%, 20%, 33%, 50%, 67% of stake)
+- Time to detect and react to a safety breach?
+- What is the marginal cost/benefit of an extra validator in the set/committee?
+
+## Fees
+
+Every transaction in the Aztec Network has a fee associated with it. The fee is payed in TST which has been bridged to L2.
+
+Transactions consume gas. There are two types of gas:
+
+- L2 gas: the cost of computation
+- DA gas: the cost of publishing/storing data
+
+When a user specifies a transaction, they provide values:
+
+- maxFeePerL2Gas: the maximum fee they are willing to pay in TST per unit L2 gas
+- maxFeePerDAGas: the maximum fee they are willing to pay in TST per unit DA gas
+- l2GasLimit: the maximum amount of L2 gas they are willing to consume
+- daGasLimit: the maximum amount of DA gas they are willing to consume
+
+Thus, the maximum fee they are willing to pay is:
+
+- maxFee = maxFeePerL2Gas _ l2GasLimit + maxFeePerDAGas _ daGasLimit
+
+There is an additional pair of parameters to support complex flow such as fee abstraction:
+
+- l2TeardownGasLimit: the maximum amount of L2 gas they are willing to consume for the teardown of the transaction
+- daTeardownGasLimit: the maximum amount of DA gas they are willing to consume for the teardown of the transaction
+
+Both of these values are used to "pre-pay" for the public teardown phase of the transaction.
+
+Each L2 block has a fixed L2 gas limit and a DA gas limit.
```

Comment:
Please can we add a note that there will be a mechanism to update / compute the required gas price, or maxFeePerDAGas and maxFeePerL2Gas that runs when a block is proven, either in solidity or inside the circuit. Exact mechanism is TBC.